File: python/Code/Course.py
from Instructor import Instructor
from Assignment import Assignment

class Course:
    # Konstruktor untuk membuat objek Course baru
    def __init__(self, code, title, instructor):
        self.__code = code
        self.__title = title
        self.__instructor = instructor
        self.__assignments = []

    # Menambahkan tugas baru ke dalam list
    def addAssignment(self, assignment):
        self.__assignments.append(assignment)

    # Mengambil daftar tugas
    def getAssignments(self):
        return self.__assignments

    # Mengambil objek instruktur
    def getInstructor(self):
        return self.__instructor

    # Mengambil judul mata kuliah
    def getTitle(self):
        return self.__title

    # Mengambil kode mata kuliah
    def getCourseCode(self):
        return self.__code

    # Mengubah kode mata kuliah
    def setCourseCode(self, code):
        self.__code = code
    
    # Mengubah judul mata kuliah
    def setTitle(self, title):
        self.__title = title

    # Mengganti instruktur mata kuliah
    def setInstructor(self, instructor):
        self.__instructor = instructor

File: java/Code/Course.java
import java.util.ArrayList;
import java.util.List;

// Kelas untuk merepresentasikan data sebuah mata kuliah (Course)
public class Course {
    // Atribut
    private String courseCode;
    private String title;
    private Instructor instructor;
    private List<Assignment> assignments;

    // Konstruktor untuk membuat objek Course baru
    public Course(String courseCode, String title, Instructor instructor) {
        this.courseCode = courseCode;
        this.title = title;
        this.instructor = instructor;
        this.assignments = new ArrayList<>();
    }

    // Mengambil kode mata kuliah
    public String getCourseCode() {
        return courseCode;
    }

    // Mengambil judul mata kuliah
    public String getTitle() {
        return title;
    }

    // Mengambil objek instruktur
    public Instructor getInstructor() {
        return instructor;
    }

    // Mengambil daftar tugas
    public List<Assignment> getAssignments() {
        return assignments;
    }

    // Mengubah kode mata kuliah
    public void setCourseCode(String courseCode) {
        this.courseCode = courseCode;
    }

    // Mengubah judul mata kuliah
    public void setTitle(String title) {
        this.title = title;
    }

    // Mengganti instruktur mata kuliah
    public void setInstructor(Instructor instructor) {
        this.instructor = instructor;
    }

    // Menambahkan tugas baru ke dalam list
    public void addAssignment(Assignment assignment) {
        this.assignments.add(assignment);
    }
}
